base/server/python/pki/server/cli/cert.py

```diff
@@ -959,30 +970,37 @@ def __init__(self):
         super(CertFixCLI, self).__init__(
             'fix', 'Fix expired system certificate(s).')
 
+    PKIDBUSER_LDIF_TEMPLATE = (
+        "dn: {dn}\n"
+        "changetype: modify\n"
+        "add: userCertificate\n"
+        "userCertificate:< file://{der_file}\n"
+    )
+
     def print_help(self):
         print('Usage: pki-server cert-fix [OPTIONS]')
         # CertID:  subsystem, sslserver, kra_storage, kra_transport, ca_ocsp_signing,
         # ca_audit_signing, kra_audit_signing
         # ca.cert.list=signing,ocsp_signing,sslserver,subsystem,audit_signing
         print()
         print('      --cert <Cert ID>            Fix specified system cert (default: all certs).')
+        print('      --extra-cert <Serial>       Also renew cert with given serial number.')
+        print('      --agent-uid <String>        UID of Dogtag agent user')
+        print('      --ldapi-socket <Path>       Path to DS LDAPI socket')
         print('  -i, --instance <instance ID>    Instance ID (default: pki-tomcat).')
-        print('  -d <NSS database>               NSS database location (default: ~/.dogtag/nssdb)')
-        print('  -c <NSS DB password>            NSS database password')
-        print('  -C <path>                       Input file containing the password for the NSS '
-              'database.')
-        print('  -n <nickname>                   Client certificate nickname')
```

Comment:
These changes will break the existing functionality in PKI 10.7 (and possibly 10.6 too) as documented here:
https://github.com/dogtagpki/pki/blob/master/docs/admin/Offline_System_Certificate_Renewal.md

Reply:
Right, the changes actually simplify things for the operator (IMO) and avoid problems with expired agent cert.  So... are you happy if I add a commit to update the documentation (I forgot it is now in-tree)?

Reply:
Hi, I just want to get some clarification. IIUC before this patch the cert-fix will work with a local and remote DS using CA agent cert authentication via LDAPS. With this patch the cert-fix will only work with a local DS via LDAPI, and the DS admin (not the CA agent) will have to enable LDAPI. Is that correct? Although it's a simpler procedure, it no longer supports certain cases used by PKI in general (i.e. remote DS).

The team had a discussion about this in the last meeting, and it's still to be determined if the remote DS support in cert-fix is required. If it is actually required, is it possible to restore the original CA agent parameters while keeping the new LDAPI parameters so the tool can support both scenarios? If that is not possible, it might more appropriate to add the LDAPI tool as an IPA tool instead (e.g. ipa-cert-fix) and keep the original pki-server cert-fix tool unchanged.

Reply:
@edewata I will make a change to support connecting to remote LDAP, but we will use Directory Manager credential instead of agent cert.  Using the DM credential supports:

- switching Dogtag from LDAPS TLS client authn to password authentication, avoiding expired subsystem cert issues
- resetting agent account password to support agent password authentication, avoiding expired agent cert issues

The only caveat will be that when using a non-local LDAP server, cert-fix cannot handle expired DS service certificate (but that was the case before all my changes, as well).

I'll update the PR, probably on Monday, so we can merge this to master at 10.6 branches.

Reply:
Still no update on actual requirement, but cfu basically confirmed that we don't have to follow a standard procedure (using CA agent) for fixing system issue like this. So feel free to implement the renewal with DM credentials. Thanks!